userguide/tutorials/payment_plugin.adoc

```diff
@@ -689,32 +702,84 @@ public class GoCardlessActivator extends KillbillActivatorBase{
++
+Some Important Pointers:
++
+** In order to ensure that the invoice balance is adjusted after updating the payment status, you need to make sure the payment goes through the internal payment control state machine. If you are using the https://github.com/killbill/killbill-plugin-framework-java[_Kill Bill plugin framework_], you can use the  https://github.com/killbill/killbill-plugin-framework-java/blob/870ae40ab476a3cc2d85b89eb2d920e4504e2784/src/main/java/org/killbill/billing/plugin/api/core/PaymentApiWrapper.java#L74-L94[_PaymentApiWrapper#transitionPendingTransaction_] method which does this for you.
++
+** Kill Bill also provides a https://killbill.github.io/slate/#payment-transaction-mark-a-pending-payment-transaction-as-succeeded-or-failed[_Mark a pending payment transaction as succeeded or failed_] endpoint. This internally invokes the https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentApi.java#L402[_PaymentApi#notifyPendingTransactionOfStateChanged_] method and can also be used.
+
+* *Via the Janitor* - As explained earlier, the https://docs.killbill.io/latest/userguide_payment.html#_janitor[__Kill Bill Janitor__] attempts to to fix *PENDING* and *UNKNOWN* payment states by querying the <<getPaymentInfo, getPaymentInfo>> method. Thus, once the plugin receives a notification from the payment gateway about the payment being processed, the `getPaymentInfo` should return the correct payment status.
+
+[NOTE]  
+*Note*: The https://docs.killbill.io/latest/userguide_payment.html#_janitor_notification_queue[notification queue janitor] runs automatically as per a configured schedule. If you would like to run the Janitor immediately, you can trigger the https://docs.killbill.io/latest/userguide_payment.html#_on_the_fly_janitor[__on the fly janitor__] by invoking the https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentApi.java#L439[_PaymentApi#getPayment_] method. 
+
+*Other Notes*
+
+Sometimes, in addition to processing the payment after a delay, the payment gateway may also charge a smaller amount then passed to it by a plugin method. The plugin should let Kill Bill know the actual processed amount so that Kill Bill can update the invoice/account balance accordingly.  In such cases, you cannot use the https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentApi.java#L402[_PaymentApi#notifyPendingTransactionOfStateChanged_] method as it can be used to only update the payment status. So, in such cases you need to rely on the Janitor and let the `getPaymentInfo` return the processed amount via the  `PaymentTransactionInfoPlugin#getAmount` method.
```

Comment:
```suggestion
Sometimes, in addition to processing the payment after a delay, the payment gateway may also charge a smaller amount then was passed to it by a plugin method. The plugin should let Kill Bill know the actual processed amount so that Kill Bill can update the invoice/account balance accordingly. In such cases, you cannot use the https://github.com/killbill/killbill-api/blob/4ae1c343a593de937415e21feecb9f5405037fa3/src/main/java/org/killbill/billing/payment/api/PaymentApi.java#L402[_PaymentApi#notifyPendingTransactionOfStateChanged_] method as it can be used only to update the payment status. So, in such cases you need to rely on the Janitor and let the `getPaymentInfo` return the processed amount via the  `PaymentTransactionInfoPlugin#getAmount` method.
```